AppCenterData/AppCenterData/Internal/Client/MSTokenExchange.m

```diff
@@ -79,7 +79,7 @@ + (void)performDbTokenAsyncOperationWithHttpClient:(id<MSHttpClientProtocol>)htt
 
             // Token store returned non-200 response code.
             if (response.statusCode != MSHTTPCodesNo200OK) {
-              MSLogError([MSData logTag], @"The token store returned %ld", (long)response.statusCode);
+              MSLogWarning([MSData logTag], @"The token store returned %ld", (long)response.statusCode);
```

Comment:
I don't have all the context of this issue so I'd hope you can help me understand these two questions:

1. Would there be any case in this codepath where 404 response from token exchange is caused by an actual error which we want to log as MSLogError? In other words, any false negatives?

2. From a design perspective, ideally we want to avoid making a call to token exchange when we know it's going to be a 404. Is there any way to filter out this scenario when we make this call in `MSTokenExchange`? I don't think there's a way to do that. Just want to confirm this is the case.

Reply:
@dhei 
1- your point is valid, reverted the change and kept the original error

2- added the change to avoid redundant call to token exchange service.